rs/execution_environment/src/execution_environment.rs

```diff
@@ -3174,36 +3218,19 @@ impl ExecutionEnvironment {
     /// Aborts paused execution in the given state.
     pub fn abort_canister(&self, canister: &mut CanisterState, log: &ReplicaLogger) {
         if !canister.system_state.task_queue.is_empty() {
-            let task_queue = std::mem::take(&mut canister.system_state.task_queue);
-            canister.system_state.task_queue = task_queue
-                .into_iter()
-                .map(|task| match task {
-                    ExecutionTask::AbortedExecution { .. }
-                    | ExecutionTask::AbortedInstallCode { .. }
-                    | ExecutionTask::Heartbeat
-                    | ExecutionTask::GlobalTimer
-                    | ExecutionTask::OnLowWasmMemory => task,
-                    ExecutionTask::PausedExecution { id, .. } => {
-                        let paused = self.take_paused_execution(id).unwrap();
-                        let (input, prepaid_execution_cycles) = paused.abort(log);
-                        self.metrics.executions_aborted.inc();
-                        ExecutionTask::AbortedExecution {
-                            input,
-                            prepaid_execution_cycles,
-                        }
-                    }
-                    ExecutionTask::PausedInstallCode(id) => {
-                        let paused = self.take_paused_install_code(id).unwrap();
-                        let (message, call_id, prepaid_execution_cycles) = paused.abort(log);
-                        self.metrics.executions_aborted.inc();
-                        ExecutionTask::AbortedInstallCode {
-                            message,
-                            call_id,
-                            prepaid_execution_cycles,
-                        }
-                    }
-                })
-                .collect();
+            let paused_tasks = canister.system_state.task_queue.get_all_paused_tasks();
+
+            self.metrics
+                .executions_aborted
+                .inc_by(paused_tasks.len().try_into().unwrap());
+
+            let aborted_tasks = self.abort_paused_executions_and_return_tasks(paused_tasks, log);
+
+            canister
+                .system_state
+                .task_queue
+                .replace_paused_with_aborted_tasks(aborted_tasks);
```

Comment:
I know that the part of:

1. taking all paused tasks from TaskQueue 
2. then aborting them in `PausedExecutionRegistry` 
3. and finally replacing paused to aborted task on the task queue 

Is not ideal but at the moment it is the best option in my knowledge if we want to make the task queue private. It is planned to refactor [`PausedExecutionRegistry`](https://dfinity.atlassian.net/browse/EXC-1730) to become private since there is no reason to be public, and at that moment we can skip the first step in this process.